cfgov/jinja2/v1/_includes/macros/video-player.html

```diff
@@ -27,27 +27,30 @@
 
 {% macro render(options={}) -%}
     {% import 'macros/share.html' as share with context %}
-    {% set video_player = options.video.player | default('youtube') %}
-    {% set video_id     = options.video.id     | default('') %}
-    {% set video_url    = options.video.url    | default('/') %}
+    {% set video_url    = options.video.url | default('/') %}
+    {% set video_player = options.video.player or
+                          ( video_url.find( 'youtube.com') > -1 and 'youtube' ) or
+                          ( video_url.find( 'ustream.tv') and 'ustream' ) %}
+    {#  TODO: Replace with django method for getting video id if youtube #}
+    {% set video_id     = options.video.id or ( video_url.split('?')[0].split('/') | last )
```

Comment:
yeah, its code that needs to get moved out of the template. Will make that change.